src/typedefs.rs

```diff
@@ -533,6 +557,33 @@ macro_rules! construct_tag {
         /// # Security:
         /// - __**Avoid using**__ `unprotected_as_bytes()` whenever possible, as it breaks all protections
         /// that the type implements.
+        ///
+        /// - The trait `PartialEq<&'_ [u8]>` is implemented for this type so that users are not tempted
+        /// to call `unprotected_as_bytes` to compare this sensitive value to a byte string. The trait
+        /// is implemented in such a way that the comparison happens in constant time. Thus, users should
+        /// prefer `SecretType == &[u8]` over `SecretType.unprotected_as_bytes() == &[u8]`.
+        /// Examples are shown below.
+        /// ```rust
+        /// use orion::hazardous::mac::hmac::Tag;
+        ///
+        /// // Initialize a tag with 64 random bytes.
+        /// let tag_bytes: [u8; 64] =
```

Comment:
To keep the examples short and concise please remove `tag_bytes` and initialize the secret key with something like `&[0u8; 32]` and compare with the same in the `assert`.